scripts/install-kubeflex.sh

```diff
@@ -0,0 +1,156 @@
+# [--strip-bin] remove the bin sub-folder
+# [-V|--verbose] verbose output
+# [-X] `set -x`
+
+set -e
+
+version=""
+os=""
+arch=""
+folder=""
+strip_bin="false"
+verbose="false"
+
+get_os_type() {
+  case "$OSTYPE" in
+      linux*)   echo "linux" ;;
+      darwin*)  echo "darwin" ;;
+      *)        echo "Unsupported operating system type: $OSTYPE" >&2 ; exit 1 ;;
+  esac
+}
+
+get_arch_type() {
+  case "$HOSTTYPE" in
+      x86_64*)  echo "amd64" ;;
+      aarch64*) echo "arm64" ;;
+      arm64*)   echo "arm64" ;;
+      *)        echo "Unsupported architecture type: $HOSTTYPE" >&2 ; exit 1 ;;
+  esac
+}
+
+get_latest_version() {
+    curl -sL https://github.com/kubestellar/kubeflex/releases/latest | grep "</h1>" | tail -n 1 | sed -e 's/<[^>]*>//g' | xargs
+}
+
+get_full_path() {
+    echo "$(cd "$1"; pwd)"
+}
+
+while (( $# > 0 )); do
+    case "$1" in
+    (--version)
+        if (( $# > 1 ));
+        then { version="$2"; shift; }
+        else { echo "$0: missing release version" >&2; exit 1; }
+        fi;;
+    (--os)
+        if (( $# > 1 ));
+        then { os="$2"; shift; }
+        else { echo "$0: missing OS type" >&2; exit 1; }
+        fi;;
+    (--arch)
+        if (( $# > 1 ));
+        then { arch="$2"; shift; }
+        else { echo "$0: missing architecture type" >&2; exit 1; }
+        fi;;
+    (--ensure-folder)
+        if (( $# > 1 ));
+        then { folder="$2"; shift; }
+        else { echo "$0: missing installation folder" >&2; exit 1; }
+        fi;;
+    (--strip-bin)
+        strip_bin="true";;
+    (--verbose|-V)
+        verbose="true";;
+    (-X)
+        set -x;;
+    (-h|--help)
+        echo "Usage: $0 [--version release] [--os linux|darwin] [--arch amd64|arm64] [--ensure-folder name] [--strip-bin] [-V|--verbose] [-X]"
+        exit 0;;
+    (-*)
+        echo "$0: unknown flag" >&2
+        exit 1;;
+    (*)
+        echo "$0: unknown positional argument" >&2
+        exit 1;;
+    esac
+    shift
+done
+
+if [ "$version" == "" ]; then
+    version=$(get_latest_version)
+fi
+
+if [ "$os" == "" ]; then
+    os=$(get_os_type)
+fi
+
+if [ "$arch" == "" ]; then
+    arch=$(get_arch_type)
+fi
+
+if [ "$folder" == "" ]; then
+    folder="$PWD/kubeflex"
+fi
+
+if [ -d "$folder" ]; then :
+else
+    if [ $verbose == "true" ]; then
+        echo "Creating folder: $folder"
+    fi
+    mkdir -p "$folder"
+fi
+
+if [ $verbose == "true" ]; then
+    echo "Downloading kubeflex $version $os/$arch..."
+    curl -SL -o kubeflex.tar.gz "https://github.com/kubestellar/kubeflex/releases/download/${version}/kubeflex_${version//v}_${os}_${arch}.tar.gz"
+else
+    curl -sSL -o kubeflex.tar.gz "https://github.com/kubestellar/kubeflex/releases/download/${version}/kubeflex_${version//v}_${os}_${arch}.tar.gz"
+fi
+
+if [ $verbose == "true" ]; then
+    echo "Extracting archive to: $folder"
+fi
+
+if [ $strip_bin == "true" ]; then
+    tar -C $folder -zxf kubeflex.tar.gz --wildcards --strip-components=1 bin/*
```

Comment:
It appears that --wildcards option is not supported on macOS. I am getting the error:

```
$ bash <(curl -s -L https://raw.githubusercontent.com/kubestellar/kubeflex/0bb3db7f20b2c98d1d057bc7fa03132330da6dc0/scripts/install-kubeflex.sh) --ensure-folder /usr/local/bin --strip-bin
tar: Option --wildcards is not supported
Usage:
  List:    tar -tf <archive-filename>
  Extract: tar -xf <archive-filename>
  Create:  tar -cf <archive-filename> [filenames...]
  Help:    tar --help
```